src/compiler/checker.ts

```diff
@@ -4738,6 +4739,14 @@ namespace ts {
             return checkTypeAssignableTo(source, target, /*errorNode*/ undefined);
         }
 
+        /**
+         * This is *not* a bi-directional relationship.

```

Comment:
It is true of `isTypeAssignableTo` but `isTypeAssignableTo` has a clear direction; on the other hand, `isTypeComparableTo` is vague enough that it warrants a comment. One would think that `A` being comparable to `B` implies that `B` is comarable to `A`, but that's not quite the case.

I *could* have made `isTypeComparableTo` a bidirectional relation, but I decided not to based on the way in which we use the relation in type assertions. Specifically, with type assertions, we check whether the expression type is comparable to the asserted type and that the asserted type is comparable to the *widened* type of the expression.

Actually, I'm not sure if there is any reason we don't check against the widened type for both. I'll experiment with it.